docs/multi-datasource/client_management_design.md

```diff
@@ -130,14 +130,14 @@ We need to configure the data source client by either creating a new one, or loo
   }
   ```
 
-- Get root client: Look up client Pool by **endpoint**, return client if existed. If misses, we create new client instance and load into pool. At this step, the client won't have any auth info.
+- Get root client: Look up client Pool by **endpoint**, return client if existed. If it misses, we create new client instance and load into pool. At this step, the client won't have any auth info.
 
 - Get credentials: Call crypto service utilities to **decrypt** user credentials from `DataSource` Object.
-- Assemble the actual query client: With auth info and root client, we’ll leverage the openearch-js connection pooling strategy to create the actual query client from root client by `client.child()`.
+- Assemble the actual query client: With auth info and root client, we’ll leverage the `opensearch-js` connection pooling strategy to create the actual query client from root client by `client.child()`.
 
 #### 4.2.1 Legacy Client
 
-OpenSearch Dashboards is forked from Kibana 7.10. At the time of the fork happened, there are 2 types of client used in the codebase. One is the new client, which later was migrated as `opensearhc-js`, the other one is the legacy client which is `elasticsearc-js`. Legacy clients are still used many critical features, such as visualization, index pattern management, along with new client.
+OpenSearch Dashboards is forked from Kibana 7.10. At the time of the fork happened, there are 2 types of client used in the codebase. One is the new client, which later was migrated as `opensearch-js`, the other one is the legacy client which is `elasticsearch-js`. Legacy clients are still used many critical features, such as visualization, index pattern management, along with new client.
```

Comment:
```suggestion
OpenSearch Dashboards was forked from Kibana 7.10 and at the time, the codebase made two types of clients available for use. One was the "new client" which has since been separated into `opensearch-js`, and the other was the legacy client named `elasticsearch-js`. Legacy clients are still used by some core features like visualization and index pattern management.
```